### PR TITLE
mesh read write

### PR DESCRIPTION
**Pull request by thewtex**
- ENH: Add itk::ReadMesh
- ENH: Add itk::WriteMesh

### CODE REVIEW

**Comment by N-Dekker** [comment]
Would be really great to have such simple read and write functions for meshes as well  😃 Could you possibly extend the tests a little bit more? I would like to see a very simple test that shows that a mesh, written by WriteMesh can be read back losslessly by ReadMesh. In pseudo code:

    itk::WriteMesh(inputMesh, fileName);
    const auto readMesh = itk::ReadMesh<MeshType>(fileName);
    // Expect the mesh read by ReadMesh to be equal to the input mesh, written by WriteMesh:
    EXPECT_EQ(*readMesh, *inputMesh);

File: Modules/IO/MeshBase/include/itkMeshFileReader.h [hunk 1]
```diff
@@ -187,6 +187,29 @@ class ITK_TEMPLATE_EXPORT MeshFileReader : public MeshSource<TOutputMesh>
...
+ * `TOutputMesh` is the expected output mesh type, and the optional
+ * `ConvertPointPixelTraits`, ``ConvertCellPixelTraits` template parameters are used to do the conversion,
+ * as specified by MeshFileReader.
+ *
+ * The function reads the mesh from the specified file, and returns the
+ * mesh that it has read.
+ * */
+template <typename TOutputMesh,
+          typename ConvertPointPixelTraits = MeshConvertPixelTraits<typename TOutputMesh::PixelType>,
+          class ConvertCellPixelTraits = MeshConvertPixelTraits<typename TOutputMesh::CellPixelType>>
```
Comment (dzenanz) [inline Modules/IO/MeshBase/include/itkMeshFileReader.h:203]: Why call one a `typename` and the other a class?

File: Modules/IO/MeshBase/test/itkMeshFileReaderWriterTest.cxx [hunk 2]
```diff
@@ -54,6 +55,9 @@ itkMeshFileReaderTest(int argc, char * argv[])
   MeshType::Pointer readMesh = nullptr;
   ITK_TRY_EXPECT_NO_EXCEPTION(readMesh = itk::ReadMesh<MeshType>(inputFileName));
 
+  ITK_TRY_EXPECT_NO_EXCEPTION(itk::WriteMesh(readMesh.GetPointer(), inputFileName));
```
Comment (dzenanz) [inline Modules/IO/MeshBase/test/itkMeshFileReaderWriterTest.cxx:58]:
```suggestion
  ITK_TRY_EXPECT_NO_EXCEPTION(itk::WriteMesh(readMesh.GetPointer(), outputFileName));
```
Also, define `outputFileName` from `argv[2]`.

**Comment by thewtex** [comment]
> I would like to see a very simple test that shows that a mesh, written by WriteMesh can be read back losslessly by ReadMesh

Yes, I added this to the test.

File: Modules/IO/MeshBase/include/itkMeshFileReader.h (hunk 1 above)
Comment (thewtex) [reply Modules/IO/MeshBase/include/itkMeshFileReader.h:203]: This is just a result of copy-paste. I will use `typename` in both places.

File: Modules/IO/MeshBase/test/itkMeshFileReaderWriterTest.cxx (hunk 2 above)
Comment (thewtex) [reply Modules/IO/MeshBase/test/itkMeshFileReaderWriterTest.cxx:58]: :+1:

File: Modules/IO/MeshBase/ITKKWStyleOverwrite.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+itkMeshFileReader\.h Comments Disable
```
Comment (N-Dekker) [inline Modules/IO/MeshBase/ITKKWStyleOverwrite.txt:1]: Thank you Matt. But why is KWStyle check disabled for "itkMeshFileReader.h"?

**Review by dzenanz — APPROVED** [review]
Probably squash that rename into the appropriate commit.

File: Modules/IO/MeshBase/ITKKWStyleOverwrite.txt (hunk 3 above)
Comment (thewtex) [reply Modules/IO/MeshBase/ITKKWStyleOverwrite.txt:1]: Hmm, possibly this was required in a previous revision -- I will remove it.

File: Modules/IO/MeshBase/test/itkMeshFileReaderWriterTest.cxx [hunk 4]
```diff
@@ -55,8 +57,19 @@ itkMeshFileReaderWriterTest(int argc, char * argv[])
   MeshType::Pointer readMesh = nullptr;
   ITK_TRY_EXPECT_NO_EXCEPTION(readMesh = itk::ReadMesh<MeshType>(inputFileName));
 
-  ITK_TRY_EXPECT_NO_EXCEPTION(itk::WriteMesh(readMesh.GetPointer(), inputFileName));
-  ITK_TRY_EXPECT_NO_EXCEPTION(itk::WriteMesh(readMesh, inputFileName));
+  const std::string outputFileName = argv[1];
```
Comment (dzenanz) [inline Modules/IO/MeshBase/test/itkMeshFileReaderWriterTest.cxx:60]: should be `argv[2]`.

**Review by dzenanz — COMMENTED** [review]
Looks good.

File: Modules/IO/MeshBase/test/itkMeshFileReaderWriterTest.cxx (hunk 4 above)
Comment (thewtex) [reply Modules/IO/MeshBase/test/itkMeshFileReaderWriterTest.cxx:60]: Good catch!

**Review by dzenanz — APPROVED** [review]

**Pull request merged**